### DISCUSSION TITLE
Design rationale behind building the Dockerfile from scratch

### DISCUSSION THREAD

**Discussion by simonlui** [Nuullll/ipex-sd-docker-for-arc-gpu#17] [General]
I hope you do not take offense at this question, but seeing as you are employed by Intel, I do find it strange that you didn't use any of the Dockerfiles provided in the Intel repositories or Docker Hub for this project directly as some sort of base to build off using a `FROM` statement in the Dockerfile. There are base Dockerfiles you could've used directly from the intel-extension-for-pytorch or intel-extension-for-tensorflow repositories at https://github.com/intel/intel-extension-for-pytorch/blob/xpu-master/docker/Dockerfile.xpu and https://github.com/intel/intel-extension-for-tensorflow/blob/main/docker/itex-xpu.Dockerfile respectively. Additionally, Intel seems to provide a base Docker image with the various OneAPI toolkits installed at https://hub.docker.com/r/intel/oneapi. It is a bit confusing to me you had these options available to you and chose none of them for this project.

**Comment by Nuullll** [comment]
Thanks for the question. I did investigate all these images before starting this project.

1. IPEX only depends on two components of oneAPI base toolkit: DPC++ and MKL (actually runtime libraries are enough). The [oneAPI images](https://hub.docker.com/r/intel/oneapi) include too much unnecessary stuff for running SD.Next. Minimizing the image size is always one of my top priorities.
2. [This dockerfile](https://github.com/Nuullll/ipex-sd-docker-for-arc-gpu/blob/main/Dockerfile) is actually derived from [IPEX xpu dockerfile](https://github.com/intel/intel-extension-for-pytorch/blob/xpu-master/docker/Dockerfile.xpu). Sadly, intel hasn't published a newer IPEX-XPU docker image -- the latest image ([tag:gpu](https://hub.docker.com/layers/intel/intel-optimized-pytorch/gpu/images/sha256-4d4b06040e9ee8ca4e5055142514b91506cf23880a630f8a912bb58ef61d016e?context=explore)) is still using oneAPI 2022.2 + IPEX-1.10.200+gpu. This is why I started to build the docker image myself: https://github.com/intel/intel-extension-for-pytorch/issues/374
3. Customized the xpu dockerfile to satisfy SD.Next's need: e.g. adding `sycl-ls`

I'm surely a docker newbie, so maybe I did this in a dumb way :-P
Do you have better advice?

**Reply by simonlui** [reply]
Ah okay, I can understand the bloat issue for the base oneAPI image. I wasn't aware though of the issues with keeping the Dockerfiles updated for IPEX and ITEX so that is a shame it isn't keep up to date every release. I think it makes sense then doing it from scratch.
If IPEX did have everything, then you could've built your image on top of it by referencing it again with a FROM statement which would've been what I expected, but it is what it is. Hopefully that changes in the future but this answers my question. Thank you for answering.

**Comment by Disty0** [comment]
Installing ipex or itex in the docker file will be useless for SD.Next since it will ignore that and create it's own environment.

And installing the whole OneAPI Basekit will unnecessarily bloat the docker image.
